tests/test_requests.py

```diff
@@ -623,6 +623,21 @@ def test_DIGESTAUTH_QUOTES_QOP_VALUE(self, httpbin):
         r = requests.get(url, auth=auth)
         assert '"auth"' in r.request.headers['Authorization']
 
+    def test_DIGESTAUTH_DOESNT_RESEND(self, httpbin):
+        """If HTTPDigestAuth fails a 401 challenge, it won't try again"""
+        s = requests.Session()
+        auth = requests.auth.HTTPDigestAuth('user', 'pass')
+
+        r = requests.Request('GET', httpbin('digest-auth/auth/user/pass'), auth=auth)
+        p = r.prepare()
+
+        # simulate previously failed 401 so auth fails
+        auth._thread_local.num_401_calls = 2
```

Comment:
So, this doesn't assert about the state of `num_401_calls`, but it does still *tweak* the state of `num_401_calls`. That is, if we removed `num_401_calls`, this test would fail. I'm still not sure that's a good idea.